internal/rest/resources/cluster.go

```diff
@@ -192,3 +206,251 @@ func clusterGet(state *state.State, r *http.Request) response.Response {
 
 	return response.SyncResponse(true, apiClusterMembers)
 }
+
+// clusterDisableMu is used to prevent the daemon process from being replaced/stopped during removal from the
+// cluster until such time as the request that initiated the removal has finished. This allows for self removal
+// from the cluster when not the leader.
+var clusterDisableMu sync.Mutex
+
+// Re-execs the daemon of the cluster member with a fresh state.
+func clusterMemberPut(state *state.State, r *http.Request) response.Response {
+	err := state.Database.Stop()
+	if err != nil {
+		return response.SmartError(fmt.Errorf("Failed shutting down database: %w", err))
+	}
+
+	err = os.RemoveAll(state.OS.StateDir)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("Failed to remove the state directory: %w", err))
+	}
+
+	go func() {
+		<-r.Context().Done() // Wait until request has finished.
+
+		// Wait until we can acquire the lock. This way if another request is holding the lock we won't
+		// replace/stop the LXD daemon until that request has finished.
+		clusterDisableMu.Lock()
+		defer clusterDisableMu.Unlock()
+		execPath, err := os.Readlink("/proc/self/exe")
+		if err != nil {
+			execPath = "bad-exec-path"
+		}
+
+		// The execPath from /proc/self/exe can end with " (deleted)" if the lxd binary has been removed/changed
+		// since the lxd process was started, strip this so that we only return a valid path.
+			logger.Info("Restarting daemon following removal from cluster")
```

Comment:
Strange tabbing here.

Reply:
Do you run your code through `go fmt` on save?

Reply:
Vim plugins were probably acting up. Sorry about that.